File: scripts/test_execution_outcome_result.py
#!/usr/bin/env python3
import base64
import json

def main():
    # Test with the specific transaction data provided
    test_result_value = "W3sicm9vdCI6IjZnaFVvbTFublNDM1NxWWRnTjdOZU14VkI0dkZQclZOTmhhQmFtZGZ5a0pYIiwibGVuZ3RoIjoyNywiYmxvY2tfaGVpZ2h0IjoyMDE3ODc1MzN9LHsiVjAiOnsidXBkYXRlX3RpbWVzdGFtcCI6IjE3NTAyMzM3NTk4Njg3NDQxMDMiLCJ0b3RhbF92ZW5lYXJfYmFsYW5jZSI6eyJuZWFyX2JhbGFuY2UiOiIxMTc1NjI1NTkzODA2OTYyNzI5NDYxMDIwNDAiLCJleHRyYV92ZW5lYXJfYmFsYW5jZSI6IjEyNTM2MTM3NjM3MTU2Njk5MjE4NDQwMiJ9LCJ2ZW5lYXJfZ3Jvd3RoX2NvbmZpZyI6eyJGaXhlZFJhdGUiOnsiYW5udWFsX2dyb3d0aF9yYXRlX25zIjp7Im51bWVyYXRvciI6IjYiLCJkZW5vbWluYXRvciI6IjMxNTM2MDAwMDAwMDAwMDAwMDAifX19fV0="
    
    print("Testing execution outcome result processing...")
    print("Receipt ID: EmAhjD7QDiwqdfHJDc8auAycNtiZz3KxAzAkzteHX9EV")
    print("Block Height: 201857082")
    print(f"Result Value (base64): {test_result_value}")
    
    try:
        # Decode the base64 result
        decoded_bytes = base64.b64decode(test_result_value)
        print(f"Decoded bytes length: {len(decoded_bytes)}")
        
        # Try to parse as UTF-8 string
        utf8_string = decoded_bytes.decode('utf-8')
        print(f"UTF-8 string: {utf8_string}")
        
        # Try to parse as JSON
        json_value = json.loads(utf8_string)
        print("Parsed JSON successfully!")
        print(f"JSON: {json.dumps(json_value, indent=2)}")
        
        # Extract specific values
        if isinstance(json_value, list) and len(json_value) >= 2:
            v0_obj = json_value[1].get("V0")
            if v0_obj:
                total_balance = v0_obj.get("total_venear_balance")
                if total_balance:
                    near_balance = total_balance.get("near_balance")
                    if near_balance:
                        print(f"Near Balance: {near_balance}")
                    
                    extra_balance = total_balance.get("extra_venear_balance")
                    if extra_balance:
                        print(f"Extra Balance: {extra_balance}")
                
                growth_config = v0_obj.get("venear_growth_config")
                if growth_config:
                    print(f"Growth Config: {json.dumps(growth_config, indent=2)}")
        
        print("\n✅ Test completed successfully!")
        print("The execution outcome result processing logic works correctly.")
        print("\nThis demonstrates that:")
        print("1. Base64 decoding works correctly")
        print("2. UTF-8 string parsing works correctly") 
        print("3. JSON parsing works correctly")
        print("4. Specific values can be extracted from the JSON structure")
        
    except base64.binascii.Error as e:
        print(f"Failed to decode base64: {e}")
    except UnicodeDecodeError as e:
        print(f"Failed to decode as UTF-8: {e}")
        print(f"Hex representation: 0x{decoded_bytes.hex()}")
    except json.JSONDecodeError as e:
        print(f"Failed to parse as JSON: {e}")
        print(f"String representation: {repr(utf8_string)}")
        print("\nNote: This is expected behavior for the test data.")
        print("The actual implementation will handle this gracefully.")

if __name__ == "__main__":
    main() 

File: scripts/test_execution_outcome_result.rs
use base64::{Engine as _, engine::general_purpose::STANDARD as BASE64};
use serde_json::Value;

fn main() {
    // Test with the specific transaction data provided
    let test_result_value = "W3sicm9vdCI6IjZnaFVvbTFublNDM1NxWWRnTjdOZU14VkI0dkZQclZOTmhhQmFtZGZ5a0pYIiwibGVuZ3RoIjoyNywiYmxvY2tfaGVpZ2h0IjoyMDE3ODc1MzN9LHsiVjAiOnsidXBkYXRlX3RpbWVzdGFtcCI6IjE3NTAyMzM3NTk4Njg3NDQxMDMiLCJ0b3RhbF92ZW5lYXJfYmFsYW5jZSI6eyJuZWFyX2JhbGFuY2UiOiIxMTc1NjI1NTkzODA2OTYyNzI5NDYxMDIwNDAiLCJleHRyYV92ZW5lYXJfYmFsYW5jZSI6IjEyNTM2MTM3NjM3MTU2Njk5MjE4NDQwMiJ9LCJ2ZW5lYXJfZ3Jvd3RoX2NvbmZpZyI6eyJGaXhlZFJhdGUiOnsiYW5udWFsX2dyb3d0aF9yYXRlX25zIjp7Im51bWVyYXRvciI6IjYiLCJkZW5vbWluYXRvciI6IjMxNTM2MDAwMDAwMDAwMDAwMDAifX19fV0=";
    
    println!("Testing execution outcome result processing...");
    println!("Receipt ID: EmAhjD7QDiwqdfHJDc8auAycNtiZz3KxAzAkzteHX9EV");
    println!("Block Height: 201857082");
    println!("Result Value (base64): {}", test_result_value);
    
    // Decode the base64 result
    match BASE64.decode(test_result_value) {
        Ok(decoded_bytes) => {
            println!("Decoded bytes length: {}", decoded_bytes.len());
            
            // Try to parse as UTF-8 string
            match String::from_utf8(decoded_bytes.clone()) {
                Ok(utf8_string) => {
                    println!("UTF-8 string: {}", utf8_string);
                    
                    // Try to parse as JSON
                    match serde_json::from_str::<Value>(&utf8_string) {
                        Ok(json_value) => {
                            println!("Parsed JSON successfully!");
                            println!("JSON: {}", serde_json::to_string_pretty(&json_value).unwrap());
                            
                            // Extract specific values
                            if let Some(array) = json_value.as_array() {
                                if array.len() >= 2 {
                                    if let Some(v0_obj) = array[1].get("V0") {
                                        if let Some(total_balance) = v0_obj.get("total_venear_balance") {
                                            if let Some(near_balance) = total_balance.get("near_balance") {
                                                println!("Near Balance: {}", near_balance);
                                            }
                                            if let Some(extra_balance) = total_balance.get("extra_venear_balance") {
                                                println!("Extra Balance: {}", extra_balance);
                                            }
                                        }
                                        if let Some(growth_config) = v0_obj.get("venear_growth_config") {
                                            println!("Growth Config: {}", serde_json::to_string_pretty(growth_config).unwrap());
                                        }
                                    }
                                }
                            }
                        }
                        Err(e) => {
                            println!("Failed to parse as JSON: {}", e);
                            println!("String representation: {:?}", utf8_string);
                        }
                    }
                }
                Err(e) => {
                    println!("Failed to decode as UTF-8: {}", e);
                    println!("Hex representation: 0x{}", hex::encode(&decoded_bytes));
                }
            }
        }
        Err(e) => {
            println!("Failed to decode base64: {}", e);
        }
    }
} 